roles/edpm_nova/tasks/configure.yml

```diff
@@ -39,6 +39,7 @@
     - {"path": "/var/lib/iscsi"}
     - {"path": "/run/openvswitch"}
 - name: Render nova config files
+  become: true
```

Comment:
this intentionally does not have become true as it should not be required if the permission are correct on the dirs so we should not proceed with this
until we identify what the regression was and fix that.

Reply:
on line 16 we are doing

  become: true
  ansible.builtin.file:
    path: "{{ item.path }}"
    state: "directory"
    setype: "container_file_t"
    owner: "{{ item.owner | default(ansible_user) }}"
    group: "{{ item.group | default(ansible_user) }}"
    mode: "{{ item.mode | default(omit) }}"
  loop:
- {"path": "{{ edpm_nova_config_dest }}", "mode": "0755"}

i.e. we are creating edpm_nova_config_dest with 755 permission and the ownership is set to the current ansible user

we should be able to rewite to edpm_nova_config_dest without become 
when we invoke ansible.builtin.template below on line 46

Reply:
@Jaganathancse can you do a `sudo grep denied /var/log/audit/audit.log` in the `edpm-compute-0` node?

Reply:
can you provide the content of the config map you are defining the custom nova.conf  and also how sriov is being configured in general.

we have not actually started testing/enabling sriov or dpdk yet.
so you are trying to do something that we don't plan to enable until at least q1 next year. we expect it to work or almost work today but no one has done it yet so its not nessirarly expected to work today.

we should not need to add become here but we may need other changes in the role to make it work

/var/lib/openstack/config/nova/config.json should be owned by the current user
so you should not be getting  a permission denied and then only role that is allowed to modify that or the /var/lib/openstack/config/nova directory in general is the nova role can you confirm that you have not modified it in any way.

please provide "ls -alZ /var/lib/openstack/config/nova"  in addition to the audit log info.

Reply:
> sudo grep denied /var/log/audit/audit.log

[cloud-admin@tigon26 ~]$ sudo grep denied /var/log/audit/audit.log
type=AVC msg=audit(1703236288.332:30): avc:  denied  { search } for  pid=2354 comm="modprobe" name="events" dev="tracefs" ino=4099 scontext=system_u:system_r:openvswitch_load_module_t:s0 tcontext=system_u:object_r:tracefs_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1703237550.644:26): avc:  denied  { search } for  pid=2111 comm="modprobe" name="events" dev="tracefs" ino=1040 scontext=system_u:system_r:openvswitch_load_module_t:s0 tcontext=system_u:object_r:tracefs_t:s0 tclass=dir permissive=0
type=AVC msg=audit(1703237550.644:26): avc:  denied  { search } for  pid=2111 comm="modprobe" name="events" dev="tracefs" ino=1040 scontext=system_u:system_r:openvswitch_load_module_t:s0 tcontext=system_u:object_r:tracefs_t:s0 tclass=dir permissive=0

Reply:
> /var/lib/openstack/config/nova/config.json
@SeanMooney 
nova config file is not generated on my env,
[cloud-admin@tigon26 ~]$ ls /var/lib/openstack/config/nova/config.json 
ls: cannot access '/var/lib/openstack/config/nova/config.json': Permission denied
[cloud-admin@tigon26 ~]$ sudo ls /var/lib/openstack/config/nova/config.json 
ls: cannot access '/var/lib/openstack/config/nova/config.json': No such file or directory

Reply:
> "ls -alZ /var/lib/openstack/config/nova

[cloud-admin@tigon26 ~]$ ls -alZ /var/lib/openstack/config/nova
ls: cannot access '/var/lib/openstack/config/nova': Permission denied

Reply:
sudo?

Reply:
> sudo?

[cloud-admin@tigon26 ~]$ sudo ls -alZ /var/lib/openstack/config/nova
total 0
drwxr-xr-x. 2 cloud-admin cloud-admin unconfined_u:object_r:container_file_t:s0  6 Dec 22 09:43 .
drwxr-x---. 5 root        root        unconfined_u:object_r:container_file_t:s0 48 Dec 22 09:43 ..

Reply:
so that is say the permisions are wrong for  /var/lib/openstack/config/nova
drwxr-x--- is not 755 its 750

Reply:
line 16 should have fixed that are you perhaps connecting as root instead of cloud-admin for some reason?

we should not be logging into the comptue as root when executing the ansibel roles

Reply:
by the way on your system if you just sudo rm -rf /var/lib/openstack/config/nova and reexecut the ansibel it should resolve your current issue

Reply:
> line 16 should have fixed that are you perhaps connecting as root instead of cloud-admin for some reason?
> 
> we should not be logging into the comptue as root when executing the ansibel roles

updated directory permission 755.

Reply:
> on line 16 we are doing
> 
> become: true ansible.builtin.file: path: "{{ item.path }}" state: "directory" setype: "container_file_t" owner: "{{ item.owner | default(ansible_user) }}" group: "{{ item.group | default(ansible_user) }}" mode: "{{ item.mode | default(omit) }}" loop:
> 
> * {"path": "{{ edpm_nova_config_dest }}", "mode": "0755"}
> 
> i.e. we are creating edpm_nova_config_dest with 755 permission and the ownership is set to the current ansible user
> 
> we should be able to rewite to edpm_nova_config_dest without become when we invoke ansible.builtin.template below on line 46

@SeanMooney 
Updated this change based on this comment.